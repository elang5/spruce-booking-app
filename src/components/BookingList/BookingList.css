ul {
  list-style-type: none;
}

.page-numbers {
  display: flex;
  justify-content: flex-end;
}

.page-numbers > li {
  text-align: center;
  width: 18px;
  margin: 0 5px;
  background-color: #f29648;
  color: white;
  padding: 5px;
  cursor: pointer;
  border-radius: 5px;
}

.page-numbers > li:last-child {
  margin-right: 20px;
}

.page-number.active {
  background-color: white;
  color: #333333;
}