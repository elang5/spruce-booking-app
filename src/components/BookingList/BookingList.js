import React, { Component } from "react"
import BookingListItem from "../BookingItem/BookingItem"
import "./BookingList.css"

export default class BookingListPage extends Component {
  static defaultProps = {
    bookingList: []
  }
  
  constructor() {
    super()
    this.state = {
      currentPage: 1,
      bookingsPerPage: 20
    }
  }

  handleClick = e => {
    this.setState({ currentPage: parseInt(e.target.id) })
  }

  renderBookingList = bookings => {
    return bookings.map((booking, index) => 
      <BookingListItem booking={booking} key={index}/>)
  }

  renderPageNumbers = pageNumbers => {
    const { currentPage } = this.state
    return pageNumbers.map(number => {
      return (
        <li
          key={number}
          id={number}
          onClick={this.handleClick}
          className={currentPage === number ? "page-number active" : "page-number"}
        >
          {number}
        </li>)
    })
  }

  render() {
    const { currentPage, bookingsPerPage } = this.state
    const { bookingList } = this.props

    const lastBookingIndex = currentPage * bookingsPerPage
    const firstBookingIndex = lastBookingIndex - bookingsPerPage
    const currentBookings = bookingList.slice(firstBookingIndex, lastBookingIndex)

    const pageNumbers = []
    for (let i = 1; i <= Math.ceil(bookingList.length / bookingsPerPage); i++) {
      pageNumbers.push(i)
    }

    return (
      <main role="main">
        <ul className="page-numbers">
          {this.renderPageNumbers(pageNumbers)}
        </ul>
        <section className="booking-list-container">
          <div className="heading cell">Customer</div>
          <div className="heading cell">Email</div>
          <div className="heading cell">Address</div>
          <div className="heading cell">Booking Type</div>
          <div className="heading cell datetime">Booking Date/Time</div>
          {this.renderBookingList(currentBookings)}
        </section>
      </main>
    )
  }
}


