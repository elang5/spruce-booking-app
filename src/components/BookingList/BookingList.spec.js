import React from 'react';
import ReactDOM from 'react-dom';
import BookingList from './BookingList';
import renderer from 'react-test-renderer';

describe('BookingList Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(< BookingList />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders the UI as expected', () => {
    const tree = renderer
      .create(<BookingList/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})
