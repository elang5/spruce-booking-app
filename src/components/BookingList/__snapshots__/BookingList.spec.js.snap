// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookingList Component renders the UI as expected 1`] = `
<main
  role="main"
>
  <ul
    className="page-numbers"
  />
  <section
    className="booking-list-container"
  >
    <div
      className="heading cell"
    >
      Customer
    </div>
    <div
      className="heading cell"
    >
      Email
    </div>
    <div
      className="heading cell"
    >
      Address
    </div>
    <div
      className="heading cell"
    >
      Booking Type
    </div>
    <div
      className="heading cell datetime"
    >
      Booking Date/Time
    </div>
  </section>
</main>
`;
